server/controllers/adminDeactivate.js

```diff
@@ -0,0 +1,122 @@
+import { User, Article } from '../models';
+
+/**
+ * @class adminDeactivate
+ * @description class to deactivate delete a user from author haven
+ */
+class AdminDeactivateController {
+  /**
+   * @description Deletes an author and update all articles to new author role
+   * @param {object} req express request object
+   * @param {object} res express response object
+   * @param {object} next next middleware
+   * @returns {next} next calls next
+   * @memberof AdminDeactivateController
+   */
+  static deleteAuthor(req, res, next) {
+    const { authorId } = req.params;
+    return User
+      .findById(authorId, {
+        attributes: ['username', 'email', 'bio'],
+        include: [{
+          model: Article,
+          as: 'authored',
+          attributes: {
+            exclude: ['timeToRead', 'createdAt', 'updatedAt']
+          }
+        }]
+      })
+      .then((user) => {
+        if (!user) {
+          return res.status(404).json({
+            error: { message: 'User does not exist' },
+            status: 'error'
+          });
+        }
+        Article
+          .update(
+            { authorId: 5 },
+            { where: { authorId } }
+          )
+          .then(() => {
+            User
+              .destroy({
+                where: { id: authorId }
+              });
+            res.status(200).json({
+              message: 'user successfully deleted',
+              status: 'success'
+            });
+          })
+          .catch(error => res.status(400).json({
```

Comment:
Nice job Gift. Although(on line 51), I think you should be consistent with one catch method. catch(next) is the convention here. Same applies to line 118